.github/workflows/build-and-publish.yml

```diff
@@ -3,12 +3,20 @@ name: Build and Publish
 on:
   workflow_call:
     inputs:
+      app_name:
+        description: "name of application folder under infra directory"
+        required: true
+        type: string
```

Comment:
Will this fail if something other than frontend or api is entered? In that case should we make this type a choice instead of string to reduce the chance of mistakes?

Reply:
It won't run if you don't manage to type the app name correctly. We have another ticket to work on monorepo structure of Github Actions after integrating backend. See #285. The infrastructure for the api doesn't exist yet, so adding `api` has a choice to deploy would fail for that reason.